pkg/loop/internal/relayer/relayer.go

```diff
@@ -71,6 +72,11 @@ func (p *PluginRelayerClient) NewRelayer(ctx context.Context, config string, key
 		}
 		deps.Add(ksCSARes)
 
+		if bc := beholder.GetClient(); bc != nil && !bc.IsSignerSet() {
+			bc.SetSigner(csaKeystore)
+			p.Logger.Debug("Set beholder signer from CSA keystore")
+		}
```

Comment:
This shouldn't be necessary on the client side. We should have already set it during startup
```suggestion
```